### PR TITLE
qa/workunits/rados/test_librados_build.sh: install build deps

### PR DESCRIPTION
**Pull request by tchaikov**
to build helloworld of librados, we need to install the build
dependencies.

Fixes: https://tracker.ceph.com/issues/40155
Signed-off-by: Kefu Chai <kchai@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by tchaikov** [comment]
this should address the test failure of http://pulpito.ceph.com/kchai-2019-06-11_01:51:33-rados-wip-kefu-testing-2019-06-09-0009-distro-basic-smithi/4021048/
```
2019-06-11T02:22:14.190 INFO:tasks.workunit.client.0.smithi088.stderr:+ check_sources
2019-06-11T02:22:14.190 INFO:tasks.workunit.client.0.smithi088.stderr:+ for s in '$SOURCES'
2019-06-11T02:22:14.191 INFO:tasks.workunit.client.0.smithi088.stderr:+ test -f hello_radosstriper.cc
2019-06-11T02:22:14.191 INFO:tasks.workunit.client.0.smithi088.stderr:+ for s in '$SOURCES'
2019-06-11T02:22:14.191 INFO:tasks.workunit.client.0.smithi088.stderr:+ test -f hello_world_c.c
2019-06-11T02:22:14.191 INFO:tasks.workunit.client.0.smithi088.stderr:+ for s in '$SOURCES'
2019-06-11T02:22:14.191 INFO:tasks.workunit.client.0.smithi088.stderr:+ test -f hello_world.cc
2019-06-11T02:22:14.191 INFO:tasks.workunit.client.0.smithi088.stderr:+ for s in '$SOURCES'
2019-06-11T02:22:14.191 INFO:tasks.workunit.client.0.smithi088.stderr:+ test -f Makefile
2019-06-11T02:22:14.192 INFO:tasks.workunit.client.0.smithi088.stderr:+ make all-system
2019-06-11T02:22:14.193 INFO:tasks.workunit.client.0.smithi088.stdout:g++ -std=c++11 -Wno-unused-parameter -Wall -Wextra -Werror -g   -o hello_world_cpp hello_world.cc -lrados -lradosstriper
2019-06-11T02:22:14.193 INFO:tasks.workunit.client.0.smithi088.stderr:make: g++: Command not found
2019-06-11T02:22:14.193 INFO:tasks.workunit.client.0.smithi088.stderr:make: *** [hello_world_cpp] Error 127
```

**Comment by smithfarm** [comment]
@tchaikov Any reason to prefer lsb_release over `source /etc/os-release` ? The latter will work fine on SUSE, while lsb_release will not.

**Comment by smithfarm** [comment]
see https://github.com/ceph/ceph/blob/master/install-deps.sh#L271 (code starting from this line) for how we are doing it in install-deps.sh

**Comment by smithfarm** [comment]
Also, could you add "Fixes: https://tracker.ceph.com/issues/40155" to the commit message?

**Comment by tchaikov** [comment]
@smithfarm thanks for the review. i will update `qa/workunits/ceph-helpers-root.sh` and `qa/workunits/rados/test_envlibrados_for_rocksdb.sh` as well.

File: qa/workunits/rados/test_librados_build.sh [hunk 1]
```diff
@@ -56,6 +73,7 @@ function run_binaries () {
 }
 
 pushd $DESTDIR
+install g++ make libradospp-dev librados-dev
```
Comment (smithfarm) [inline qa/workunits/rados/test_librados_build.sh]: On SUSE (and - I believe - also in RHEL/CentOS), the RPM containing `/usr/bin/g++` is called "gcc-c++", not "g++". The others should be OK.

Comment (tchaikov) [reply qa/workunits/rados/test_librados_build.sh]: ack. will fix this in my next revision!

**Comment by tchaikov** [comment]
retest this please.

File: qa/workunits/rados/test_librados_build.sh [hunk 2]
```diff
@@ -56,6 +58,14 @@ function run_binaries () {
 }
 
 pushd $DESTDIR
+case $(distro_id) in
+    centos|fedora|rhel)
+        install gcc-c++.x86_64 make libradospp-devel librados-devel;;
```
Comment (smithfarm) [inline qa/workunits/rados/test_librados_build.sh]: Please add a new clause here:

```
opensuse*|suse|sles)
    install gcc-c++ make libradospp-devel librados-devel;;
```

File: qa/workunits/rados/test_envlibrados_for_rocksdb.sh [hunk 3]
```diff
@@ -40,15 +19,15 @@ CURRENT_PATH=`pwd`
...
-case $(lsb_release -si) in
-	Ubuntu|Debian|Devuan)
+case $(distro_id) in
+	ubuntu|debian|devuan)
 		install g++ libsnappy-dev zlib1g-dev libbz2-dev libradospp-dev
 		;;
-	CentOS|Fedora|RedHatEnterpriseServer)
+	centos|fedora|rhel)
 		install gcc-c++.x86_64 snappy-devel zlib zlib-devel bzip2 bzip2-devel libradospp-devel.x86_64
 		;;
```
Comment (smithfarm) [inline qa/workunits/rados/test_envlibrados_for_rocksdb.sh]: Please add a new clause here:

```
opensuse*|suse|sles)
    install gcc-c++ snappy-devel zlib-devel libbz2-devel libradospp-devel
    ;;
```

File: qa/workunits/ceph-helpers-root.sh [hunk 4]
```diff
@@ -17,26 +17,30 @@
...
+    case $(distro_id) in
+        ubuntu|debian|devuan)
+            sudo env DEBIAN_FRONTEND=noninteractive apt-get install -y "$@"
             ;;
-        CentOS|Fedora|RedHatEnterpriseServer)
+        centos|fedora|rhel)
             sudo yum install -y "$@"
             ;;
-        *SUSE*)
+        *suse*)
```
Comment (smithfarm) [inline qa/workunits/ceph-helpers-root.sh]: please change this line to:

```
opensuse*|suse|sles)
```

Comment (tchaikov) [reply qa/workunits/ceph-helpers-root.sh]: done.

File: qa/workunits/rados/test_envlibrados_for_rocksdb.sh (hunk 3 above)
Comment (tchaikov) [reply qa/workunits/rados/test_envlibrados_for_rocksdb.sh]: this is not in the scope of this commit. will do so in a separated commit.

File: qa/workunits/rados/test_librados_build.sh (hunk 2 above)
Comment (tchaikov) [reply qa/workunits/rados/test_librados_build.sh]: done.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-06-12_09:47:11-rados-wip-xio-cleanup-kefu-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2019-06-12_09:48:15-rados-wip-xio-cleanup-kefu-distro-basic-smithi/

**Pull request merged**